src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -302,6 +301,16 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         {
             StorageApplicationPermissions::FutureAccessList().AddOrReplace(L"PickedFolderToken", folder);
             _State.Profile().StartingDirectory(folder.Path());
+            StartingDirectoryUseParentCheckbox().IsChecked(false);
+        }
+    }
+
+    void Profiles::StartingDirectory_LostFocus(IInspectable const&, RoutedEventArgs const&)
+    {
+        if (StartingDirectory().Text().empty())
+        {
+            StartingDirectoryUseParentCheckbox().IsChecked(true);
+            StartingDirectory().IsEnabled(false);
         }
     }
```

Comment:
prefer subscribing to the profile view model's PropertyChanged notification and waiting for StartingDirectory to change, _then_ checking the real live value of `_Profile.StartingDirectory()`. That will eliminate false positives related to the textbox focus change.